docs/content/config/advanced/kubernetes.md

```diff
@@ -4,518 +4,626 @@ title: 'Advanced | Kubernetes'
-              volumeMounts:
-                - name: config
-                  subPath: postfix-main.cf
-                  mountPath: /tmp/docker-mailserver/postfix-main.cf
-                  readOnly: true
-                - name: config
-                  subPath: postfix-master.cf
-                  mountPath: /tmp/docker-mailserver/postfix-master.cf
-                  readOnly: true
-                - name: config
-                  subPath: dovecot.cf
-                  mountPath: /tmp/docker-mailserver/dovecot.cf
-                  readOnly: true
-    ```
+          entryPoints: [ submissions ]
+          routes:
+            - match: HostSNI(`*`)
+              services:
+                - name: mailserver
+                  namespace: mail
+                  port: subs-proxy # note the 15 character limit here
+                  proxyProtocol:
+                    version: 2
+
+        ---
+        apiVersion: traefik.io/v1alpha1
+        kind: IngressRouteTCP
+
+        metadata:
+          name: imaps
 
-With this approach,
+        spec:
+          entryPoints: [ imaps ]
+          routes:
+            - match: HostSNI(`*`)
+
+              services:
+                - name: mailserver
+                  namespace: mail
+                  port: imaps-proxy
+                  proxyProtocol:
+                    version: 2
+        ```
+
+        The `*-proxy` ports that we refer to in the `IngressRouteTCP`s are configured on the `mailserver` service, and these ports refer to the `Deployment`'s ports again. One has two options for configuring the `mailserver` service now:
+
+        1. In case you do not need cluster-internal e-mails to reach DMS on default ports, you can simply change existing port configurations to use the PROXY protocol.
+        2. In case you do need (or want) cluster-internal e-mails to reach DMS on default ports, you need to duplicate port configurations in order to open PROXY-protocol-aware ports and non-PROXY-protocol-aware ports.
+
+        === "Cluster-Internal E-Mails Not Required"
+
+            Here is an exmaple configuration for [Postfix][docs-postfix], [Dovecot][docs-dovecot], and the `Deployment`:
+
+            ```yaml
+            kind: ConfigMap
+            apiVersion: v1
+            metadata:
+              name: mailserver-extra-config
+              labels:
+                app: mailserver
+            data:
+              postfix-main.cf: |
+                postscreen_upstream_proxy_protocol = haproxy
+              postfix-master.cf: |
+                smtp/inet/postscreen_upstream_proxy_protocol=haproxy
+                submission/inet/smtpd_upstream_proxy_protocol=haproxy
+                submissions/inet/smtpd_upstream_proxy_protocol=haproxy
+              dovecot.cf: |
+                haproxy_trusted_networks = <YOUR POD CIDR>, 127.0.0.0/8
+                service imap-login {
+                  inet_listener imap {
+                    haproxy = yes
+                  }
```

Comment:
You're not configuring `imap-proxy` below, why is this here?

What is with the `127.0.0.0/8` network for `haproxy_trusted_networks`? DMS isn't speaking PROXY protocol to itself on loopback? Was this actually required for something? It's not present in the separate ports config tab or in either of the dedicated PROXY protocol docs page examples.

I'm still not a fan of an entire subnet as is encouraged here with the pod CIDR, but I understand that's a k8s concern. You just have to trust that there is no networking surprises that permit traffic to slip through and inject their own PROXY protocol header.

Reply:
> You're not configuring `imap-proxy` below, why is this here?

Can you elaborate? I am not sure if I understand.

> What is with the `127.0.0.0/8` network for `haproxy_trusted_networks`? DMS isn't speaking PROXY protocol to itself on loopback? Was this actually required for something? It's not present in the separate ports config tab or in either of the dedicated PROXY protocol docs page examples.

We can remove the 127.0.0.1/8, it serves no purpose.

> I'm still not a fan of an entire subnet as is encouraged here with the pod CIDR, but I understand that's a k8s concern. You just have to trust that there is no networking surprises that permit traffic to slip through and inject their own PROXY protocol header.

It's actually required because the ingress pod IP can change at any point of the pod is destroyed (for whatever reason). Using the Pod-CIDR here is absolutely correct.